### ISSUE TITLE
move `MachinesByCreationTimestamp` to `util/collections`

### ISSUE THREAD

**Issue by ykakarap** [kubernetes-sigs/cluster-api#5091]
**Description**
Move `MachinesByCreationTimestamp` out of `util` package and make it a private struct in `util/collections` package.

/kind cleanup
/help
/good-first-issue

**Comment by ykakarap** [comment]
@Priyankasaggu11929 Will you be able to work on this?

**Comment by Priyankasaggu11929** [comment]
Hello @ykakarap, apologies for leaving the issue hanging. I just got stuck in too many things together. Have unassigned myself from the issue. I'll start fresh (& pick some myself). Sorry once again & thank you! :)

**Comment by shivanshu1333** [comment]
Would like to work on it
/assign

**Comment by ykakarap** [comment]
@jayesh-srivastava @shivanshu1333 Awesome to see such interest! 🚀 
I see that both of you assigned yourself to the issue. Please coordinate on who works on this. :) I dont think it really needs 2 people working on it.

**Comment by jayesh-srivastava** [comment]
Yeah I was keen to work on this so I assigned it to myself first. So if @shivanshu1333 can `unassign`, that would be great :)

**Comment by shivanshu1333** [comment]
Sure go ahead :)
/unassign

**Comment by sandipanpanda** [comment]
@shivanshu1333 @shivanshu1333 I see that both you are already K8s Org members and have multiple previous contributions to K8s. As this  is a `good-first-issue`, I would suggest you keep this issue for someone who is a first time contributor to K8s. 
That's what good first issue means, right? :wink:

**Comment by palnabarun** [comment]
/remove-lifecycle stale
(with all the activity on the thread)

**Comment by fabriziopandini** [comment]
/unassign @jayesh-srivastava 
you can eventually pick it up again if you have the bandwidth
